### PR TITLE
e2e: set RUST_BACKTRACE in e2e tests and handle missing git commit data

### PR DESCRIPTION
**Pull request by vrutkovs**
* Update template to support RUST_BACKTRACE=1 (set to 0 by default) for
both containers
* Install `git` in Dockerfile.build so that `built_info::GIT_VERSION` would be set
* Handle missing commit info by setting "unknown" version

### CODE REVIEW

**Comment by vrutkovs** [comment]
>level=error msg="Error: Provider produced inconsistent result after apply"

/retest

File: dist/openshift/cincinnati.yaml [hunk 1]
```diff
@@ -131,6 +136,11 @@ objects:
                     configMapKeyRef:
                       key: pe.mandatory_client_parameters
                       name: cincinnati
+                - name: "RUST_BACKTRACE"
+                  valueFrom:
+                    configMapKeyRef:
+                      key: pe.rust_backtrace
+                      name: cincinnati
```
Comment (LalatenduMohanty) [inline dist/openshift/cincinnati.yaml:143]: This is temporary right?

**Comment by vrutkovs** [comment]
Returns "unknown" build commit

File: dist/Dockerfile.deploy/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
 FROM quay.io/app-sre/cincinnati:builder AS builder
 # build
 COPY . .
+# copy git information for built crate
+COPY .git/ ./.git/
```
Comment (LalatenduMohanty) [inline dist/Dockerfile.deploy/Dockerfile:8]: Why is this required?  because when you copy a git repository in  the .git directory should get copied correctly in to `../.git/`.

Comment (vrutkovs) [reply dist/Dockerfile.deploy/Dockerfile:8]: `COPY . .` doesn't copy hidden files or dirs

File: dist/openshift/cincinnati.yaml (hunk 1 above)
Comment (vrutkovs) [reply dist/openshift/cincinnati.yaml:143]: This is required for PR e2e. This won't be set on stage or prod

**Comment by vrutkovs** [comment]
/hold cancel

e2e fixed
